strawberry/schema/schema.py

```diff
@@ -259,11 +267,16 @@ class Query:
                 graphql_types.append(graphql_type)
 
         try:
+            directives = specified_directives + tuple(graphql_directives)
+
+            if self.config.enable_experimental_incremental_execution:
+                directives = tuple(directives) + tuple(incremental_execution_directives)
```

Comment:
we need to throw errors if user what experimental execution but we are not on graphql core 3.3+

Reply:
There are two `raise RuntimeError` below, which check that. Maybe it should be here during `__init__` instead, and in there just `assert`?